authority/authorize.go

```diff
@@ -2,64 +2,35 @@ package authority
 
 import (
 	"crypto/x509"
-	"encoding/asn1"
 	"net/http"
-	"net/url"
 	"time"
 
 	"github.com/pkg/errors"
-	"github.com/smallstep/cli/crypto/x509util"
-	"gopkg.in/square/go-jose.v2/jwt"
+	"github.com/smallstep/certificates/authority/provisioner"
+	"github.com/smallstep/cli/jose"
 )
 
 type idUsed struct {
 	UsedAt  int64  `json:"ua,omitempty"`
 	Subject string `json:"sub,omitempty"`
 }
 
-// Claims extends jwt.Claims with step attributes.
+// Claims extends jose.Claims with step attributes.
 type Claims struct {
-	jwt.Claims
-	SANs []string `json:"sans,omitempty"`
-}
-
-// matchesAudience returns true if A and B share at least one element.
-func matchesAudience(as, bs []string) bool {
-	if len(bs) == 0 || len(as) == 0 {
-		return false
-	}
-
-	for _, b := range bs {
-		for _, a := range as {
-			if b == a || stripPort(a) == stripPort(b) {
-				return true
-			}
-		}
-	}
-	return false
-}
-
-// stripPort attempts to strip the port from the given url. If parsing the url
-// produces errors it will just return the passed argument.
-func stripPort(rawurl string) string {
-	u, err := url.Parse(rawurl)
-	if err != nil {
-		return rawurl
-	}
-	u.Host = u.Hostname()
-	return u.String()
+	jose.Claims
+	SANs  []string `json:"sans,omitempty"`
+	Email string   `json:"email,omitempty"`
+	Nonce string   `json:"nonce,omitempty"`
 }
 
 // Authorize authorizes a signature request by validating and authenticating
 // a OTT that must be sent w/ the request.
-func (a *Authority) Authorize(ott string) ([]interface{}, error) {
-	var (
-		errContext = map[string]interface{}{"ott": ott}
-		claims     = Claims{}
-	)
+// TODO(mariano): protection against reuse for oidc
```

Comment:
is this still a TODO? is this a blocker?

Reply:
It was already done. Using a nonce in `step oauth`. 
Removing comment.